src/core/config/config.rs

```diff
@@ -68,6 +68,11 @@ pub struct Config {
     #[serde(default, skip_serializing_if = "is_default")]
     pub unions: BTreeMap<String, Union>,
 
+    ///
+    /// A map of all the interface types in the schema.
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub interfaces: BTreeMap<String, Interface>,
```

Comment:
from Config perspective any type that is used as base for other type is interface. Please check https://github.com/tailcallhq/tailcall/blob/d68f44cc9477db78fc3cec4e373fe5b2f3afb5d3/src/core/config/config_module.rs#L82

I'm not sure we need this change to the config - interface types could be checked from the config_module